core/shared/src/main/scala/zio/ZLayer.scala

```diff
@@ -2053,6 +2071,22 @@ object ZLayer {
      */
     def update[A: Tagged](f: A => A)(implicit ev: ROut <:< Has[A]): ZLayer[RIn, E, ROut] =
       self >>> ZLayer.fromFunctionMany(_.update[A](f))
+
+    /**
+     * Proves that this layer has a single `Has[_]` output,
+     * which allows more operations to be done lawfully, for example map a value inside `Has[_]`.
+     */
+    def singleService(implicit ev: HasSingleService[ROut]): SingleServiceOutputLayerOps[RIn, E, ev.Out] =
```

Comment:
I feel like HasSingleService is redundant, does it work if replaced by
```

def singleService[A](implicit ev: ROut =:= Has[A]): SingleServiceOutputLayerOps[RIn, E, A] =
```
?

Reply:
I tried it at first, and it works, but you have to annotate the call, like this:
```
layer.singleServie[Int].map(...)
```
It might be okay to be explicit what single service you want to get, but to me it seems redundant.
So this whole typeclass with a type member thing is to avoid this annotation.

Reply:
You're right, very odd, but `=:=` doesn't work without ascription